tests/Functional/Driver/PDO/MySQL/DeadlockTest.php

```diff
@@ -0,0 +1,81 @@
+<?php
+
+namespace Doctrine\DBAL\Tests\Functional\Driver\PDO\MySQL;
+
+use Doctrine\DBAL\Connection;
+use Doctrine\DBAL\Exception\DeadlockException;
+use Doctrine\DBAL\Schema\Table;
+use Doctrine\DBAL\Tests\FunctionalTestCase;
+use Doctrine\DBAL\Tests\TestUtil;
+
+use function pcntl_fork;
+use function sleep;
+
+class DeadlockTest extends FunctionalTestCase
+{
+    protected function setUp(): void
+    {
+        parent::setUp();
+
+        if (TestUtil::isDriverOneOf('pdo_mysql', 'mysqli')) {
+            $this->prepareDatabase();
+
+            return;
+        }
+
+        self::markTestSkipped('This test requires the pdo_mysql driver.');
+    }
+
+    private function prepareDatabase(): void
+    {
+        $table = new Table('test1');
+        $table->addColumn('id', 'integer', ['autoincrement' => true]);
```

Comment:
Why does the column need to be auto-incremented? It could be just nullable, or an arbitrary value could be inserted explicitly.